core/unix/signal_macos.c

```diff
@@ -157,8 +157,9 @@ sigcontext_to_mcontext_simd(priv_mcontext_t *mc, sig_full_cxt_t *sc_full)
         return;
     mc->fpsr = fpc->__fpsr;
     mc->fpcr = fpc->__fpcr;
-    ASSERT(sizeof(mc->simd) == sizeof(fpc->__v));
-    memcpy(&mc->simd, &fpc->__v, sizeof(mc->simd));
+    for (int i = 0; i < proc_num_simd_registers(); i++) {
+        memcpy(&mc->simd[i], &fpc->__v[i], sizeof(fpc->__v[i]));
```

Comment:
Add `ASSERT(sizeof(mc->simd[i]) == sizeof(fpc->__v[i]))` before this line.

Reply:
see below comment

Reply:
Use:
```
memcpy(&mc->simd[i].q, &fpc->__v[i], sizeof(mc->simd->q));
```
here.